### ISSUE TITLE
Manjaro KDE: switch to nvidia and then see blacksreen

### ISSUE THREAD

**Issue by codetang-2417** [Askannz/optimus-manager#570]
**Describe the bug**

I use the Manjaro system, which has always been working normally. However, after a recent update, when switching to the NVIDIA graphics card, the entire system goes black, but I can still switch to other TTYs. 

I checked the switch log and found: ERROR: Cannot setup PRIME (xrandr error). 

How should I troubleshoot this error? Thank you for any responses.

**System info**
Please include :
- Your distribution
```
$ neofetch
██████████████████  ████████   ling@ling-20ym 
██████████████████  ████████   -------------- 
██████████████████  ████████   OS: Manjaro Linux x86_64 
██████████████████  ████████   Host: 20YM Lenovo ThinkBook 16p Gen 2 
████████            ████████   Kernel: 6.1.94-1-MANJARO 
████████  ████████  ████████   Uptime: 8 mins 
████████  ████████  ████████   Packages: 1780 (pacman) 
████████  ████████  ████████   Shell: bash 5.2.26 
████████  ████████  ████████   Resolution: 2560x1600 
████████  ████████  ████████   DE: Plasma 6.0.5 
████████  ████████  ████████   WM: KWin 
████████  ████████  ████████   Theme: [Plasma], Breeze [GTK2/3] 
████████  ████████  ████████   Icons: [Plasma], McMojave-circle-dark [GTK2/3] 
████████  ████████  ████████   Terminal: konsole 
                               CPU: AMD Ryzen 7 5800H with Radeon Graphics (16) @ 3.200GHz 
                               GPU: AMD ATI Radeon Vega Series / Radeon Vega Mobile Series 
                               GPU: NVIDIA GeForce RTX 3060 Mobile / Max-Q 
                               Memory: 6526MiB / 23392MiB
```

- Your desktop manager (KDE, GNOME, Cinnamon...), if you have one
KDE
- You display manager (SDDM, GDM, LightDM...), if you have one. If you do not know, you are probably using the one coming with your desktop manager.
SDDM
- Your laptop model, if you think it could be related to the problem
- The version of optimus-manager you are using : latest stable release (optimus-manager), or the latest Git version (optimus-manager-git)
Optimus Manager (Client) version 1.5
optimus-manager-git 1:r729.32048d4.python-1
- Your custom optimus-manager configuration file at `/etc/optimus-manager/optimus-manager.conf`, if you made one
```
cat /etc/optimus-manager/optimus-manager.conf                                                                                                   ✔ 
[amd]
DRI=3
driver=modesetting
tearfree=

[intel]
DRI=3
accel=
driver=modesetting
modeset=yes
tearfree=

[nvidia]
DPI=96
PAT=yes
allow_external_gpus=yes
dynamic_power_management=no
dynamic_power_management_memory_threshold=0
ignore_abi=no
modeset=yes
options=

[optimus]
auto_logout=yes
pci_power_control=yes
pci_remove=no
pci_reset=hot_reset
startup_auto_battery_mode=integrated
startup_auto_extpower_mode=nvidia
startup_mode=hybrid
switching=bbswitch
```
**Logs**
Run `optimus-manager --status` in a console, and if an error message appears, post it here. The message should also point you to a log path, attach it here as well. If there is no error message but you are still experiencing issues, grab the most recent files in `/var/log/optimus-manager/switch/` and `/var/log/optimus-manager/daemon/`.

```
$ optimus-manager --status
Optimus Manager (Client) version 1.5

Current GPU mode : hybrid
GPU mode requested for next login : no change
GPU at startup : hybrid
Temporary config path: no

$ cat /var/log/optimus-manager/switch/switch-20240705T225535.log
[9] INFO: # Xorg pre-start hook
[9] INFO: Previous state was: {'type': 'pending_pre_xorg_start', 'requested_mode': 'nvidia', 'current_mode': 'hybrid'}
[9] INFO: Requested mode is: nvidia
[52] INFO: Writing to /etc/X11/xorg.conf.d/10-optimus-manager.conf
[52] INFO: Writing state {'type': 'pending_post_xorg_start', 'switch_id': '20240705T225535', 'requested_mode': 'nvidia'}
[52] INFO: Xorg pre-start hook completed successfully.
[8] INFO: # Xorg post-start hook
[34] INFO: Running xrandr commands
[71] ERROR: Cannot setup PRIME (xrandr error):
None
[71] INFO: Running /etc/optimus-manager/xsetup-nvidia.sh
[78] ERROR: Cannot set DPI (xrandr error):
None
[78] INFO: Writing state {'type': 'done', 'switch_id': '20240705T225535', 'current_mode': 'nvidia'}
[78] INFO: Xorg post-start hook completed successfully.
[9] INFO: # Xorg pre-start hook
[9] INFO: Previous state was: {'type': 'done', 'switch_id': '20240705T225535', 'current_mode': 'nvidia'}
[9] INFO: Requested mode is: nvidia
[49] INFO: Writing to /etc/X11/xorg.conf.d/10-optimus-manager.conf
[49] INFO: Writing state {'type': 'pending_post_xorg_start', 'switch_id': '20240705T225535', 'requested_mode': 'nvidia'}
[50] INFO: Xorg pre-start hook completed successfully.
```

**Comment by nwildner** [comment]
Could this be related to?

 - #115 

What happens if you follow the same procedure that is explained on that case? - : set `switching=none` and `pci_power_control=yes. To switch GPUs, use `--set-startup` + reboot instead of `--switch`.

That is for testing to check out if `switching=bbswitch` isn't playing a role here...

Also, keep using the `optimus-manager-git` since it bundles the last merged PRs from this repo :)

**Comment by codetang-2417** [comment]
Thank you for your response. I tried your suggested solution, but it didn't improve the situation. Before this upgrade, everything was working fine. I will continue using optimus-manager-git in hopes that future updates will resolve the issue. Additionally, would you like me to provide any relevant logs or debugging information to help the team address these types of problems?

**Comment by es20490446e** [comment]
What I do in these situations is to try different configurations till I know exactly which component or setting is failing.

For example, you can try:

1. What happens if you use a different driver like `nvidia-open` or the nvidia proprietary one?

2. And if you edit `optimus-manager.conf` to only have:

```
# This is the user configuration file for optimus-manager.
# The default configuration file is at /usr/share/optimus-manager.conf.
# Any option specified in this file will override the default configuration.
# When overriding an option, remember to include the corresponding section header.

[optimus]
startup_mode=nvidia
```

Then progressively add your original settings back.

Keep discarding options till you know which one specifically is causing the issue. Then come back, reopen this report, let us know what you have discovered, and we can look into the code.

If it helps I'm using also Manjaro KDE Plasma on GeForce 1650 and I'm not affected. I'm using `nvidia-open` with the conf above.

**Comment by codetang-2417** [comment]
I think I can now provide some useful information:
Previously, when I was testing, I only used the laptop's built-in screen. Now I've connected an external monitor and obtained the following results:

The external monitor and built-in display work well in hybrid mode.
Problems occurred when switching to NVIDIA-only mode: the external monitor works normally, but the built-in display goes black. Additionally, the Optimus Manager log shows "xrandr is not working properly". Then, I tried to enable the built-in screen directly through the command line using 'xrandr --output eDP-1-1 --auto', but received an error: xrandr: Configure crtc 4 failed. So I believe the problem is with xrandr and not related to Optimus. I apologize for raising an unrelated issue with you.

Thank you very much for your responses!
